### DISCUSSION TITLE
How do I allow PC to wait when reading from latest offset?

### DISCUSSION THREAD

**Discussion by jaddooo** [confluentinc/parallel-consumer#643] [Q&A]
If there are no messages currently coming through the topic, how do I get the PC to continue listening for messages?

**Comment by rkolesnev** [comment]
Hi @jaddooo ,
I am not sure what you mean by that? 
Same as "plain" Kafka Consumer - Parallel Consumer will poll / listen for messages indefenitely until its explicitly closed / terminated.

**Comment by astubbs** [comment]
@jaddooo - @rkolesnev's answer stands: with the callback API, `poll(...)` keeps listening until you close the instance, and `auto.offset.reset=latest` on a new group just means nothing arrives until something new is produced. If that is what you were using, nothing was wrong.

There is one way an older version could genuinely look like it had stopped waiting, and if it is what you hit it is fixed: the `JStreamParallelStreamProcessor` result `Stream` used to end the first time its queue was momentarily empty - usually before the first record on a quiet topic - so a loop over it returned immediately and the caller walked away while PC carried on consuming behind it. On [v0.6.0.0](https://github.com/astubbs/parallel-consumer/releases/tag/v0.6.0.0) of the maintained fork that stream blocks until the consumer closes (astubbs/parallel-consumer#116, this tracker's #912); consume it on its own thread, as the Vert.x example now does.

The coordinates changed with the fork - on Maven Central it is now:

```xml
<dependency>
    <groupId>bz.stub.parallelconsumer</groupId>
    <artifactId>parallel-consumer-core</artifactId>
    <version>0.6.0.0</version>
</dependency>
```

(`parallel-consumer-vertx`, `parallel-consumer-reactor` and `parallel-consumer-mutiny` likewise.) The Java package is `bz.stub.parallelconsumer` too, so imports change with it - one command rewrites them (GNU `sed`; on macOS use `sed -i ''`):

```bash
find . -name '*.java' -exec sed -i 's/io\.confluent\.parallelconsumer/bz.stub.parallelconsumer/g' {} +
```
